### PR TITLE
feat(cognito): support `emailVerified` for `AttributeMapping` interface

### LINKED ISSUE #30467: aws-cognito: UserPoolIdentityProviderOidc does not support `email_verified` as attributeMapping
### Describe the bug

Hi all,

I'm configuring a custom OpenId provide on my cognito user pool. I'd like to auto_verified emails since the openId provider give me this information.

I noticed that in the AWS console I can map the `email_verified` field on a field that my OpenId provided provides to me but looks like that the `email_verified` field is not expose as a property of the construct UserPoolIdentityProviderOidc under `attributeMapping` key.

I expect to use this code:

```
new cognito.UserPoolIdentityProviderOidc(this, 'custom',{
        name: '...',
        userPool: this.userPool,
        clientId: '...',
        attributeMapping: {
          email: cognito.ProviderAttribute.other('email'),
          email_verified: cognito.ProviderAttribute.other('email_verified'),
      }
    )
```

### Expected Behavior

I expect that using this field the email is automatically verified at the signup time

### Current Behavior

Actually this field is not available as property of the object `attributeMapping`

### Reproduction Steps

Using this code:
```
new cognito.UserPoolIdentityProviderOidc(this, 'custom',{
        name: '...',
        userPool: this.userPool,
        clientId: '...',
        attributeMapping: {
          email: cognito.ProviderAttribute.other('email'),
          email_verified: cognito.ProviderAttribute.other('email_verified'),
      }
    )
```

Typescript notifies that `email_verified` is not a know property.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.1628.0

### Framework Version

_No response_

### Node.js Version

18.18.2

### OS

MacOs 14.5

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by yasuaki640**
### Issue #30467 

Closes #30467

### Reason for this change

For custom OpenId providers, there is no way to automatically validate email upon sign-in.
Therefore, we would like to add the `email_verified` attribute to attribute mapping, but it is not present in the member  definition of `AttributeMapping` interface., so we have added it in this PR.

### Description of changes

Added `emailVerified` attribute to `AttributeMapping` interface.

### Description of how you validated changes

Added the `email_verified` assertion to both unit and integration tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by yasuaki640** [comment]
Exemption Request: Not all attributes defined in the `AttributeMapping` interface are listed in the README, so I don't think it is necessary to add them to the README.

File: packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/base.test.ts [hunk 1]
```diff
@@ -39,13 +39,15 @@ describe('UserPoolIdentityProvider', () => {
         attributeMapping: {
           givenName: ProviderAttribute.FACEBOOK_NAME,
           birthdate: ProviderAttribute.FACEBOOK_BIRTHDAY,
+          emailVerified: ProviderAttribute.other('email_verified'),
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/base.test.ts:42]: This is not a good user experience as users would need to manually figure out the exact text for email verified. I would recommend explicitly building something like 
```
public static readonly EMAIL_VERIFIED = new ProviderAttribute('email_verified');
```

**Review by GavinZZ — CHANGES_REQUESTED** [review]
Looks good, just a slight improvement. Also I would recommend adding a small section in README. You're absolutely right that the current README doesn't include every new feature, and that's because we missed it in previous code reviews. We would ideally still want to insist on adding a relevant README section for each new feature.

File: packages/aws-cdk-lib/aws-cognito/lib/user-pool-idps/base.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ import { IUserPool } from '../user-pool';
 export class ProviderAttribute {
   /** The email attribute provided by Apple */
   public static readonly APPLE_EMAIL = new ProviderAttribute('email');
+  /** The email verified atribute provided by Apple */
+  public static readonly APPLE_EMAIL_VERIFIED = new ProviderAttribute('email_verified');
```
Comment (yasuaki640) [inline packages/aws-cdk-lib/aws-cognito/lib/user-pool-idps/base.ts:10]: reference.
https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_rest_api/authenticating_users_with_sign_in_with_apple#3383773

File: packages/aws-cdk-lib/aws-cognito/lib/user-pool-idps/base.ts [hunk 3]
```diff
@@ -51,6 +53,8 @@ export class ProviderAttribute {
   public static readonly GOOGLE_PHONE_NUMBERS = new ProviderAttribute('phoneNumbers');
   /** The email attribute provided by Google */
   public static readonly GOOGLE_EMAIL = new ProviderAttribute('email');
+  /** The email verified attribute provided by Google */
+  public static readonly GOOGLE_EMAIL_VERIFIED = new ProviderAttribute('email_verified');
```
Comment (yasuaki640) [inline packages/aws-cdk-lib/aws-cognito/lib/user-pool-idps/base.ts:57]: reference.
https://developers.google.com/identity/openid-connect/openid-connect#an-id-tokens-payload

File: packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/base.test.ts (hunk 1 above)
Comment (yasuaki640) [reply packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/base.test.ts]: @GavinZZ 
Added constants for when idp is either apple or google.

**Comment by yasuaki640** [comment]
@GavinZZ 

`test/aws-cognito/test/integ.user-pool-idp.apple.ts` integration test has a dummy key set, but I don't have it.
How should I handle this test?

```shell
$ integ-runner --language javascript test/aws-cognito/test/integ.user-pool-idp.apple.js --update-on-failed

Verifying integration test snapshots...

  CHANGED    aws-cognito/test/integ.user-pool-idp.apple 1.45s
      Resources
[~] AWS::Cognito::UserPoolIdentityProvider apple9B5408AC 
 └─ [~] AttributeMapping
     └─ [+] Added: .email_verified



Snapshot Results: 

Tests:    1 failed, 1 total
Failed: /Users/yasuaki640/ghq/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js

Running integration tests for failed tests...

Running in parallel across regions: us-east-1, us-east-2, us-west-2
Running test /Users/yasuaki640/ghq/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js in us-east-1

✨  Synthesis time: 1.24s

integ-user-pool-idp-apple: deploying... [1/1]
integ-user-pool-idp-apple: creating CloudFormation changeset...
integ-user-pool-idp-apple | 0/5 | 4:22:30 PM | REVIEW_IN_PROGRESS   | AWS::CloudFormation::Stack             | integ-user-pool-idp-apple User Initiated
integ-user-pool-idp-apple | 0/5 | 4:22:37 PM | CREATE_IN_PROGRESS   | AWS::CloudFormation::Stack             | integ-user-pool-idp-apple User Initiated
integ-user-pool-idp-apple | 0/5 | 4:22:40 PM | CREATE_IN_PROGRESS   | AWS::Cognito::UserPool                 | pool (pool056F3F7E) 
integ-user-pool-idp-apple | 0/5 | 4:22:42 PM | CREATE_IN_PROGRESS   | AWS::Cognito::UserPool                 | pool (pool056F3F7E) Resource creation Initiated
integ-user-pool-idp-apple | 1/5 | 4:22:42 PM | CREATE_COMPLETE      | AWS::Cognito::UserPool                 | pool (pool056F3F7E) 
integ-user-pool-idp-apple | 1/5 | 4:22:42 PM | CREATE_IN_PROGRESS   | AWS::Cognito::UserPoolIdentityProvider | apple (apple9B5408AC) 
integ-user-pool-idp-apple | 1/5 | 4:22:42 PM | CREATE_IN_PROGRESS   | AWS::Cognito::UserPoolDomain           | pool/domain (pooldomain430FA744) 
integ-user-pool-idp-apple | 1/5 | 4:22:44 PM | CREATE_IN_PROGRESS   | AWS::Cognito::UserPoolDomain           | pool/domain (pooldomain430FA744) Resource creation Initiated
integ-user-pool-idp-apple | 1/5 | 4:22:44 PM | CREATE_FAILED        | AWS::Cognito::UserPoolIdentityProvider | apple (apple9B5408AC) Provided private key cannot be used for Sign in with Apple. (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidParameterException; Request ID: 373a191f-4d9f-4045-a677-9499b98a1cc1; Proxy: null)
integ-user-pool-idp-apple | 1/5 | 4:22:44 PM | CREATE_FAILED        | AWS::Cognito::UserPoolDomain           | pool/domain (pooldomain430FA744) Resource creation cancelled
integ-user-pool-idp-apple | 1/5 | 4:22:44 PM | ROLLBACK_IN_PROGRESS | AWS::CloudFormation::Stack             | integ-user-pool-idp-apple The following resource(s) failed to create: [pooldomain430FA744, apple9B5408AC]. Rollback requested by user.
integ-user-pool-idp-apple | 1/5 | 4:22:46 PM | DELETE_IN_PROGRESS   | AWS::Cognito::UserPoolDomain           | pool/domain (pooldomain430FA744) 
integ-user-pool-idp-apple | 2/5 | 4:22:46 PM | DELETE_COMPLETE      | AWS::Cognito::UserPoolIdentityProvider | apple (apple9B5408AC) 
integ-user-pool-idp-apple | 3/5 | 4:22:47 PM | DELETE_COMPLETE      | AWS::Cognito::UserPoolDomain           | pool/domain (pooldomain430FA744) 
integ-user-pool-idp-apple | 3/5 | 4:22:47 PM | DELETE_IN_PROGRESS   | AWS::Cognito::UserPool                 | pool (pool056F3F7E) 
integ-user-pool-idp-apple | 2/5 | 4:22:48 PM | DELETE_COMPLETE      | AWS::Cognito::UserPool                 | pool (pool056F3F7E) 
integ-user-pool-idp-apple | 3/5 | 4:22:49 PM | ROLLBACK_COMPLETE    | AWS::CloudFormation::Stack             | integ-user-pool-idp-apple 

Failed resources:
integ-user-pool-idp-apple | 4:22:44 PM | CREATE_FAILED        | AWS::Cognito::UserPoolIdentityProvider | apple (apple9B5408AC) Provided private key cannot be used for Sign in with Apple. (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidParameterException; Request ID: 373a191f-4d9f-4045-a677-9499b98a1cc1; Proxy: null)
❌  integ-user-pool-idp-apple failed: The stack named integ-user-pool-idp-apple failed creation, it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE: Provided private key cannot be used for Sign in with Apple. (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidParameterException; Request ID: ***; Proxy: null)
  FAILED     aws-cognito/test/integ.user-pool-idp.apple-integ.user-pool-idp.apple (undefined/us-east-1) 43.073s
      Integration test failed: Error: Command exited with status 1

Test Results: 

Tests:    1 failed, 1 total
Error: Some integration tests failed!
    at main (/Users/yasuaki640/ghq/github.com/aws/aws-cdk/packages/@aws-cdk/integ-runner/lib/cli.js:166:23)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by GavinZZ** [comment]
Let me help you with the tests.

**Comment by GavinZZ** [comment]
@yasuaki640 can you allow me to push to your forked repo?

**Comment by GavinZZ** [comment]
Alternatively, you can copy & paste the following diff into a file called `patch.diff` and run `git apply patch.diff`.

```
diff --git a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/cdk.out b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/cdk.out
index 588d7b269d..c6e612584e 100644
--- a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/cdk.out
+++ b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/cdk.out
@@ -1 +1 @@
-{"version":"20.0.0"}
\ No newline at end of file
+{"version":"38.0.1"}
\ No newline at end of file
diff --git a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ-user-pool-idp-apple.assets.json b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ-user-pool-idp-apple.assets.json
index b6d9e5304f..4ff567fa73 100644
--- a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ-user-pool-idp-apple.assets.json
+++ b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ-user-pool-idp-apple.assets.json
@@ -1,7 +1,7 @@
 {
-  "version": "20.0.0",
+  "version": "38.0.1",
   "files": {
-    "d435562230aa834a5fb53b4a894ffb5ef1c788cf50bcf38e5638c32afdd96619": {
+    "f3adbf3426efedfef748320d2b748e9fc0d285a04c6368f366c80422cc14e54c": {
       "source": {
         "path": "integ-user-pool-idp-apple.template.json",
         "packaging": "file"
@@ -9,7 +9,7 @@
       "destinations": {
         "current_account-current_region": {
           "bucketName": "cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}",
-          "objectKey": "d435562230aa834a5fb53b4a894ffb5ef1c788cf50bcf38e5638c32afdd96619.json",
+          "objectKey": "f3adbf3426efedfef748320d2b748e9fc0d285a04c6368f366c80422cc14e54c.json",
           "assumeRoleArn": "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/cdk-hnb659fds-file-publishing-role-${AWS::AccountId}-${AWS::Region}"
         }
       }
diff --git a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ-user-pool-idp-apple.template.json b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ-user-pool-idp-apple.template.json
index e4b7781041..9713c418de 100644
--- a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ-user-pool-idp-apple.template.json
+++ b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ-user-pool-idp-apple.template.json
@@ -34,9 +34,6 @@
   "poolclient2623294C": {
    "Type": "AWS::Cognito::UserPoolClient",
    "Properties": {
-    "UserPoolId": {
-     "Ref": "pool056F3F7E"
-    },
     "AllowedOAuthFlows": [
      "implicit",
      "code"
@@ -57,7 +54,10 @@
       "Ref": "apple9B5408AC"
      },
      "COGNITO"
-    ]
+    ],
+    "UserPoolId": {
+     "Ref": "pool056F3F7E"
+    }
    }
   },
   "pooldomain430FA744": {
@@ -72,14 +72,10 @@
   "apple9B5408AC": {
    "Type": "AWS::Cognito::UserPoolIdentityProvider",
    "Properties": {
-    "ProviderName": "SignInWithApple",
-    "ProviderType": "SignInWithApple",
-    "UserPoolId": {
-     "Ref": "pool056F3F7E"
-    },
     "AttributeMapping": {
      "family_name": "lastName",
-     "given_name": "firstName"
+     "given_name": "firstName",
+     "email_verified": "email_verified"
     },
     "ProviderDetails": {
      "client_id": "com.amzn.cdk",
@@ -87,6 +83,11 @@
      "key_id": "CDKKEYCDK1",
      "private_key": "PRIV_KEY_CDK",
      "authorize_scopes": "email name"
+    },
+    "ProviderName": "SignInWithApple",
+    "ProviderType": "SignInWithApple",
+    "UserPoolId": {
+     "Ref": "pool056F3F7E"
     }
    }
   }
diff --git a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ.json b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ.json
index 8631ca2321..dbd9d4e97a 100644
--- a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ.json
+++ b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/integ.json
@@ -1,5 +1,5 @@
 {
-  "version": "20.0.0",
+  "version": "38.0.1",
   "testCases": {
     "integ.user-pool-idp.apple": {
       "stacks": [
diff --git a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/manifest.json b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/manifest.json
index 67d15c5698..9a1344dae0 100644
--- a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/manifest.json
+++ b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/manifest.json
@@ -1,12 +1,6 @@
 {
-  "version": "20.0.0",
+  "version": "38.0.1",
   "artifacts": {
-    "Tree": {
-      "type": "cdk:tree",
-      "properties": {
-        "file": "tree.json"
-      }
-    },
     "integ-user-pool-idp-apple.assets": {
       "type": "cdk:asset-manifest",
       "properties": {
@@ -20,10 +14,12 @@
       "environment": "aws://unknown-account/unknown-region",
       "properties": {
         "templateFile": "integ-user-pool-idp-apple.template.json",
+        "terminationProtection": false,
         "validateOnSynth": false,
+        "notificationArns": [],
         "assumeRoleArn": "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/cdk-hnb659fds-deploy-role-${AWS::AccountId}-${AWS::Region}",
         "cloudFormationExecutionRoleArn": "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/cdk-hnb659fds-cfn-exec-role-${AWS::AccountId}-${AWS::Region}",
-        "stackTemplateAssetObjectUrl": "s3://cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}/d435562230aa834a5fb53b4a894ffb5ef1c788cf50bcf38e5638c32afdd96619.json",
+        "stackTemplateAssetObjectUrl": "s3://cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}/f3adbf3426efedfef748320d2b748e9fc0d285a04c6368f366c80422cc14e54c.json",
         "requiresBootstrapStackVersion": 6,
         "bootstrapStackVersionSsmParameter": "/cdk-bootstrap/hnb659fds/version",
         "additionalDependencies": [
@@ -83,6 +79,12 @@
         ]
       },
       "displayName": "integ-user-pool-idp-apple"
+    },
+    "Tree": {
+      "type": "cdk:tree",
+      "properties": {
+        "file": "tree.json"
+      }
     }
   }
 }
\ No newline at end of file
diff --git a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/tree.json b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/tree.json
index a78668bd21..ae45970e06 100644
--- a/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/tree.json
+++ b/packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-idp.apple.js.snapshot/tree.json
@@ -4,14 +4,6 @@
     "id": "App",
     "path": "",
     "children": {
-      "Tree": {
-        "id": "Tree",
-        "path": "Tree",
-        "constructInfo": {
-          "fqn": "constructs.Construct",
-          "version": "10.1.85"
-        }
-      },
       "integ-user-pool-idp-apple": {
         "id": "integ-user-pool-idp-apple",
         "path": "integ-user-pool-idp-apple",
@@ -53,7 +45,7 @@
                   }
                 },
                 "constructInfo": {
-                  "fqn": "@aws-cdk/aws-cognito.CfnUserPool",
+                  "fqn": "aws-cdk-lib.aws_cognito.CfnUserPool",
                   "version": "0.0.0"
                 }
               },
@@ -67,9 +59,6 @@
                     "attributes": {
                       "aws:cdk:cloudformation:type": "AWS::Cognito::UserPoolClient",
                       "aws:cdk:cloudformation:props": {
-                        "userPoolId": {
-                          "Ref": "pool056F3F7E"
-                        },
                         "allowedOAuthFlows": [
                           "implicit",
                           "code"
@@ -90,17 +79,20 @@
                             "Ref": "apple9B5408AC"
                           },
                           "COGNITO"
-                        ]
+                        ],
+                        "userPoolId": {
+                          "Ref": "pool056F3F7E"
+                        }
                       }
                     },
                     "constructInfo": {
-                      "fqn": "@aws-cdk/aws-cognito.CfnUserPoolClient",
+                      "fqn": "aws-cdk-lib.aws_cognito.CfnUserPoolClient",
                       "version": "0.0.0"
                     }
                   }
                 },
                 "constructInfo": {
-                  "fqn": "@aws-cdk/aws-cognito.UserPoolClient",
+                  "fqn": "aws-cdk-lib.aws_cognito.UserPoolClient",
                   "version": "0.0.0"
                 }
               },
@@ -121,19 +113,19 @@
                       }
                     },
                     "constructInfo": {
-                      "fqn": "@aws-cdk/aws-cognito.CfnUserPoolDomain",
+                      "fqn": "aws-cdk-lib.aws_cognito.CfnUserPoolDomain",
                       "version": "0.0.0"
                     }
                   }
                 },
                 "constructInfo": {
-                  "fqn": "@aws-cdk/aws-cognito.UserPoolDomain",
+                  "fqn": "aws-cdk-lib.aws_cognito.UserPoolDomain",
                   "version": "0.0.0"
                 }
               }
             },
             "constructInfo": {
-              "fqn": "@aws-cdk/aws-cognito.UserPool",
+              "fqn": "aws-cdk-lib.aws_cognito.UserPool",
               "version": "0.0.0"
             }
           },
@@ -147,14 +139,10 @@
                 "attributes": {
                   "aws:cdk:cloudformation:type": "AWS::Cognito::UserPoolIdentityProvider",
                   "aws:cdk:cloudformation:props": {
-                    "providerName": "SignInWithApple",
-                    "providerType": "SignInWithApple",
-                    "userPoolId": {
-                      "Ref": "pool056F3F7E"
-                    },
                     "attributeMapping": {
                       "family_name": "lastName",
-                      "given_name": "firstName"
+                      "given_name": "firstName",
+                      "email_verified": "email_verified"
                     },
                     "providerDetails": {
                       "client_id": "com.amzn.cdk",
@@ -162,17 +150,22 @@
                       "key_id": "CDKKEYCDK1",
                       "private_key": "PRIV_KEY_CDK",
                       "authorize_scopes": "email name"
+                    },
+                    "providerName": "SignInWithApple",
+                    "providerType": "SignInWithApple",
+                    "userPoolId": {
+                      "Ref": "pool056F3F7E"
                     }
                   }
                 },
                 "constructInfo": {
-                  "fqn": "@aws-cdk/aws-cognito.CfnUserPoolIdentityProvider",
+                  "fqn": "aws-cdk-lib.aws_cognito.CfnUserPoolIdentityProvider",
                   "version": "0.0.0"
                 }
               }
             },
             "constructInfo": {
-              "fqn": "@aws-cdk/aws-cognito.UserPoolIdentityProviderApple",
+              "fqn": "aws-cdk-lib.aws_cognito.UserPoolIdentityProviderApple",
               "version": "0.0.0"
             }
           },
@@ -180,20 +173,44 @@
             "id": "SignInLink",
             "path": "integ-user-pool-idp-apple/SignInLink",
             "constructInfo": {
-              "fqn": "constructs.Construct",
-              "version": "10.1.85"
+              "fqn": "aws-cdk-lib.CfnOutput",
+              "version": "0.0.0"
+            }
+          },
+          "BootstrapVersion": {
+            "id": "BootstrapVersion",
+            "path": "integ-user-pool-idp-apple/BootstrapVersion",
+            "constructInfo": {
+              "fqn": "aws-cdk-lib.CfnParameter",
+              "version": "0.0.0"
+            }
+          },
+          "CheckBootstrapVersion": {
+            "id": "CheckBootstrapVersion",
+            "path": "integ-user-pool-idp-apple/CheckBootstrapVersion",
+            "constructInfo": {
+              "fqn": "aws-cdk-lib.CfnRule",
+              "version": "0.0.0"
             }
           }
         },
+        "constructInfo": {
+          "fqn": "aws-cdk-lib.Stack",
+          "version": "0.0.0"
+        }
+      },
+      "Tree": {
+        "id": "Tree",
+        "path": "Tree",
         "constructInfo": {
           "fqn": "constructs.Construct",
-          "version": "10.1.85"
+          "version": "10.3.0"
         }
       }
     },
     "constructInfo": {
-      "fqn": "constructs.Construct",
-      "version": "10.1.85"
+      "fqn": "aws-cdk-lib.App",
+      "version": "0.0.0"
     }
   }
 }
\ No newline at end of file
```

**Pull request merged**